src/service.rs

```diff
@@ -393,9 +403,77 @@ where
     Ok(Body::from_json(&f().await?)?.into())
 }
 
+async fn do_reset(mnemonic: SecretString, password: SecretString, config: &Config) -> State {
```

Comment:
Please use proper error handling for this function. Don't just expect on all the errors, pass them up to the caller.